spring-boot-autoconfigure/src/test/java/org/axonframework/springboot/legacyjpa/JpaEventStoreAutoConfigurationWithSnapshottingTest.java

```diff
@@ -0,0 +1,247 @@
+import org.axonframework.commandhandling.CommandHandler;
+import org.axonframework.commandhandling.gateway.CommandGateway;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.EventHandler;
+import org.axonframework.eventsourcing.*;
+import org.axonframework.eventsourcing.eventstore.EventStore;
+import org.axonframework.eventsourcing.eventstore.legacyjpa.JpaEventStorageEngine;
+import org.axonframework.eventsourcing.snapshotting.SnapshotFilter;
+import org.axonframework.modelling.command.AggregateIdentifier;
+import org.axonframework.modelling.command.AggregateLifecycle;
+import org.axonframework.modelling.command.TargetAggregateIdentifier;
+import org.axonframework.serialization.xml.CompactDriver;
+import org.axonframework.spring.stereotype.Aggregate;
+import org.axonframework.springboot.autoconfig.*;
+import org.axonframework.springboot.autoconfig.legacyjpa.JpaJavaxAutoConfiguration;
+import org.axonframework.springboot.autoconfig.legacyjpa.JpaJavaxEventStoreAutoConfiguration;
+import org.axonframework.tracing.NoOpSpanFactory;
+import org.axonframework.tracing.SpanFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+/**
+ * Test class validating auto configured snapshotting logic.
+ *
+ * @author Steven van Beelen
+ */
+class JpaEventStoreAutoConfigurationWithSnapshottingTest {
+
+    private static final String AGGREGATE_ID = "some-aggregate";
+
+    @BeforeEach
+    void setUp() {
+        TestContext.SNAPSHOT_FILTER_INVOKED.set(false);
+    }
+
+    @Test
+    void snapshotterAndSnapshotTriggerDefinitionAreInvoked() {
+        new ApplicationContextRunner()
+                .withUserConfiguration(DataSourceAutoConfiguration.class)
+                .withUserConfiguration(JpaAutoConfiguration.class)
+                .withUserConfiguration(HibernateJpaAutoConfiguration.class)
+                .withUserConfiguration(TransactionAutoConfiguration.class)
+                .withUserConfiguration(InfraConfiguration.class)
+                .withUserConfiguration(EventProcessingAutoConfiguration.class)
+                .withUserConfiguration(JpaJavaxAutoConfiguration.class)
+                .withUserConfiguration(JpaJavaxEventStoreAutoConfiguration.class)
+                .withUserConfiguration(TestContext.class)
```

Comment:
If I recall correctly, if you annotated the `TestContext` with `@EnableAutoConfiguration`, you shouldn't be required to add the framework's auto-config classes as user configuration anymore. Might be a little cleaner to look at.

Reply:
Other more recent auto-configuration test classes of ours also replace the `@Configuration` annotation for `@ContextConfiguration`, by the way. If just adding `@EnableAutoConfiguration`, perhaps you also need to make that adjustment.

Reply:
Thanks, I did need to add `.withPropertyValues("axon.axonserver.enabled=false")` but it's much cleaner now.